### PR TITLE
fix(ext/http): Throwing Error if the return value of Deno.serve handler is not a Response class

### PR DESCRIPTION
**Pull request by codesculpture**
Fixing this https://github.com/denoland/deno/issues/20947 and #20013
We are not checking the return value of handler passed through Deno.serve and directly passing it to toInnerResponse but thats expecting a Response class . So throwing error if handler return other than Response class. (Btw, its my first PR correct me if im wrong on anything :) )

### CODE REVIEW

File: ext/http/00_serve.js [hunk 1]
```diff
@@ -449,6 +450,11 @@ function mapToCallback(context, callback, onError) {
         fromInnerRequest(innerRequest, signal, "immutable"),
         new ServeHandlerInfo(innerRequest),
       );
+      
+      // Throwing Error if the handler return value is not a Response class
+      if(!(response && ObjectPrototypeIsPrototypeOf(ResponsePrototype, response))) {
```
Comment (mmastrac) [inline ext/http/00_serve.js:455]: This doesn't need a null check as ObjectPrototypeIsPrototypeOf will return false here:

```suggestion
      if (!ObjectPrototypeIsPrototypeOf(ResponsePrototype, response)) {
```

**Comment by mmastrac** [comment]
Changes look good, but we should add a test to `serve_test.ts` to validate it.

**Comment by codesculpture** [comment]
> Changes look good, but we should add a test to `serve_test.ts` to validate it.

Okay its night here, I would write a test tmrw and will let u know @mmastrac. Thanks

File: ext/http/00_serve.js [hunk 2]
```diff
@@ -449,15 +450,26 @@ function mapToCallback(context, callback, onError) {
...
+      // Throwing Error if the handler return value is not a Response class
+      if (!ObjectPrototypeIsPrototypeOf(ResponsePrototype, response)) {
+        throw TypeError(
+          "Return value from serve handler must be a response or a promise resolving to a response",
+        );
+      }
     } catch (error) {
       try {
         response = await onError(error);
+        if (!ObjectPrototypeIsPrototypeOf(ResponsePrototype, response)) {
```
Comment (codesculpture) [inline ext/http/00_serve.js:463]: Added same logic to restrict to return only Response class in onError callback

**Comment by mmastrac** [comment]
Looks great to me. If you mark it as ready for review we can give it a run with CI!

**Comment by codesculpture** [comment]
Just doing linting and formatting (got some problem with that), and am running cargo test. Then i will mark this as read for review.

**Comment by codesculpture** [comment]
Hey @mmastrac, when i run cargo test some were failing the log is so big, but i mostly saw integration_test were failing. Is that something i need to do since i dint do any breaking change i believe

**Comment by mmastrac** [comment]
@codesculpture Sometimes we get test flakes -- let's see what the CI says.

**Comment by codesculpture** [comment]
Seems it was unexpectedly failed. Isnt it?

**Comment by mmastrac** [comment]
I can take a look! I'll try to merge it today

**Review by mmastrac — APPROVED** [review]
LGTM!

**Comment by codesculpture** [comment]
Thank you @mmastrac

**Pull request merged**